vcsim/main.go

```diff
@@ -199,6 +200,13 @@ func main() {
 
 	sig := make(chan os.Signal, 1)
 	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+	if *stdinExit {
+		fmt.Fprintf(out, "Press any key to exit")
+		go func() {
+			os.Stdin.Read(make([]byte, 1))
```

Comment:
your tty is probably going to line buffer this, isn't it? aka, the message should probably read: "Press enter to continue" or you probably need to bring in kr/pty.

Reply:
setting the tty to raw mode will probably also work, but doesn't seem like a good idea from my POV. Just wanted to make sure it was also mentioned.

Reply:
Good catch! You are probably right.

Reply:
Honestly, if you were to gather my full opinion on this...

^C is suitably fine for this and sends SIGINT, which is already trapped. If anything, perhaps an informative message when stdio is not *closed* that the user can press ^C to exit instead of "any key" or even waiting for enter to be pressed.

The simplest way to do this is just to print that message to stdout; but if you want to check if the tty is actually in use, you can validate trivially with golang.org/x/crypto/ssh/terminal IIRC.

https://godoc.org/golang.org/x/crypto/ssh/terminal#IsTerminal

Reply:
then you don't need a flag at all.

Reply:
If you know you're on a unix system, you can shell out to `tty` (/usr/bin/tty on ubuntu) which will print "not a tty" if the library import is an issue.

Other than that you can break into the syscalls that the above linked call uses... kind of a lot of work to print an error message though.